spec/lib/services/nodes-api-service-spec.js

```diff
@@ -227,25 +186,268 @@ describe("Http.Services.Api.Nodes", function () {
+                type: 'compute',
+                relations: [
+                    {
+                        "relationType": "enclosedBy",
+                        "targets": [
+                            "1234abcd1234abcd1234abf",
+                        ]
+                    }
+                ]
+            };
+
+            var pduNodeAfter = _.cloneDeep(pduNode);
+            _.pull(pduNodeAfter.relations[0].targets, pduNodeAfter.relations[0].targets[0]);
+            var computeNodeBefore = _.cloneDeep(computeNode);
+            computeNodeBefore.relations[1] = {
+                "relationType": "poweredBy",
+                "targets": ["1234abcd1234abcd1234abcg"]
+            };
+
+            getActiveTaskGraph.resolves('');
+            needByIdentifier.withArgs(computeNode.id).resolves(computeNodeBefore);
+            needByIdentifier.withArgs(computeNode2.id).resolves(computeNode2);
+            needByIdentifier.withArgs(pduNode.id).resolves(pduNode);
+            updateByIdentifier.withArgs(pduNode.id).resolves(pduNodeAfter);
+
+            return nodeApiService.removeNode(enclosureNode)
+            .then(function (node){
+                expect(node).to.equal(enclosureNode);
+                expect(waterline.nodes.destroy).to.have.been.calledTrice;
+                expect(updateByIdentifier).to.have.been
+                    .calledWith(pduNode.id,
+                                {relations: pduNodeAfter.relations});
+            });
+        });
+
+        it("removeNode should not delete enlosure when active workflow", function() {
+            var computeNode2 = {
+                id: '1234abcd1234abcd1234abce',
+                type: 'compute',
+                relations: [
+                    {
+                        "relationType": "enclosedBy",
+                        "targets": [
+                            "1234abcd1234abcd1234abcf"
+                        ]
+                    }
+                ]
+            };
+            var computeNodeAfter = _.cloneDeep(computeNode);
+            var computeNode2After = _.cloneDeep(computeNode2);
+            delete computeNodeAfter.relations;
+            delete computeNode2After.relations;
+
+            getActiveTaskGraph.resolves('');
+            getActiveTaskGraph.withArgs({target: computeNode2.id}).resolves('1');
+            needByIdentifier.withArgs(computeNode.id).resolves(computeNode);
+            needByIdentifier.withArgs(computeNode2.id).resolves(computeNode2);
+            updateByIdentifier.withArgs(computeNode.id).resolves(computeNodeAfter);
+            updateByIdentifier.withArgs(computeNode2.id).resolves(computeNode2After);
+
+            return nodeApiService.removeNode(enclosureNode)
+            .catch(function (err){
+                expect(err.name).to.equal('BadRequestError');
+                expect(waterline.nodes.destroy).to.not.have.been.called;
+                expect(updateByIdentifier).to.not.have.been.called;
+            });
```

Comment:
This is a chance that if really the removeNode resolves not rejects, then all assertion in the "catch" block will not be executed, then the test case passes because it doesn't do any assertion. We'd better check the removeNode is actually rejects, or to explicitly fail the test is resolved:

return xxx.removeNode()
.catch() {
xxxx
}.then() {
   return Promise.rejects('xxx');
}